### ISSUE TITLE
add `ext/FooExt/src/FooExt.jl` to the entry point list of an extention

### ISSUE THREAD

**Issue by ctarn** [JuliaLang/julia#48985]
Currently (1.9.0-rc1) the entry point of a package extention `FooExt` is defined as "either in ext/FooExt.jl or ext/FooExt/FooExt.jl". Would it be a good idea to add `ext/FooExt/src/FooExt.jl` to the entry point list so that an extension can also be organized and used as an individual package?

**Comment by StefanKarpinski** [comment]
cc @KristofferC

**Comment by KristofferC** [comment]
I guess. If people think it is useful it is easy to add and it doesn't really cost anything so why not?

**Comment by Roger-luo** [comment]
One question would be can this extension have its own environment? This is probably not a blocker of supporting this feature alone. 

But If it can then should Pkg load that environment and merge it into the caller? If we don't support this then we will need to add the same group of package to the weak deps field of the main package.

**Comment by KristofferC** [comment]
> One question would be can this extension have its own environment?

No, this would purely be a code organization change, where you can put the extension file.

**Comment by StefanKarpinski** [comment]
Is the motivation for this that you might want to split the extension code across multiple files?

**Comment by KristofferC** [comment]
> Is the motivation for this that you might want to split the extension code across multiple files?

You can already do this by putting the files under `ext/FooExt/`. So I guess having the `src` there would make it look a bit more like a normal package. But I don't really see a point to it other than that.

**Comment by ctarn** [comment]
To my knowledge, an extension would be loaded or executed instead of `import`ed. For example, if I load `FooExt` to `MainPkg`, I can't refer 
to the extension module using something like `MainPkg.FooExt`, and calling functions/variables defined in the extension is also not possible.

It seems that an extension should be pretty lightweight. It is mainly designed to add methods to functions defined in the main package, do some initialization, or add properties to the main package using the new feature `setproperty!`.
I enjoy using extensions. However, personally, I also think that the ability of extensions is limited.

Actually, for the original problem I met, I have found an alternative solution. However I still believe the issue should be a good feature to be considered. So let's further show the problem/motivation.

Specifically, I have a _big_ package `MainPkg` containing many functionalities. And I also have several plotting packages doing the similar things but based on different backends. Previously I can not merge the plotting functions into the main package since it would result in a heavy dependency list.

But with the extension scheme, I can now merge the plotting packages into main package. It makes my workspace much clearer.
However, it also leads to some problems. I can not easily access some backend-specific things; and sometimes I can not select an specified backend.
For example, I have two plotting packages `FooMakie.jl` and `FooPlots.jl`.

```
module FooMakie
import Makie
fooplot(args...) = ...
fooplot!(ax::Makie.Axis, args...) = ...
end

module FooPlots
import Plots
fooplot(args...) = ...
fooplot!(ax::Plots.Plot, args...) = ...
end
```
The two packages can be organized as extensions as below.
```
module MainPkg
function fooplot end
function fooplot! end
end

module FooMakieExt
import MainPkg
import Makie
MainPkg.fooplot(args...) = ...
MainPkg.fooplot!(ax::Makie.Axis, args...) = ...
end

module FooPlotsExt
import MainPkg
import Plots
MainPkg.fooplot(args...) = ...
MainPkg.fooplot!(ax::Plots.Plot, args...) = ...
end
```
The function `fooplot!` is always perfectly ok, but `fooplot` not.
When both extensions are loaded, a conflict occurs since `FooMakie.fooplot` and `FooPlots.fooplot` share the same argument list. So I would like to organize the modules also as individual `import`-able packages not just extension modules, so that I can call specific functions in such situations.
